### PR TITLE
Use new ts equality function to reduce test code

### PR DESCRIPTION
**Pull request by hyanwong**
Should be a trivial improvement

### CODE REVIEW

**Review by benjeffery — APPROVED** [review]
Nice!

**Review by jeromekelleher — CHANGES_REQUESTED** [review]
LGTM, but we do actually lose some information about where the things differ.

I've opened an issue on that to track that https://github.com/tskit-dev/tskit/issues/1076

File: tests/test_inference.py [hunk 1]
```diff
@@ -1053,14 +1038,14 @@ def test_from_historical_tree_sequence(self):
...
-class TestThreads(TsinferTestCase):
+class TestThreads:
     def test_equivalance(self):
         rho = 2
         ts = msprime.simulate(5, mutation_rate=2, recombination_rate=rho, random_seed=2)
         sample_data = tsinfer.SampleData.from_tree_sequence(ts)
         ts1 = tsinfer.infer(sample_data, num_threads=1)
         ts2 = tsinfer.infer(sample_data, num_threads=5)
-        self.assertTreeSequencesEqual(ts1, ts2)
+        ts1.equals(ts2, ignore_provenance=True)
```
Comment (jeromekelleher) [inline tests/test_inference.py:1048]: Missing an assert here, this test doesn't actually check anything now.

Comment (hyanwong) [reply tests/test_inference.py:1048]: Ha. Good spot. Sorry!

**Review by jeromekelleher — APPROVED** [review]

**Pull request merged**